src/algorithms/all_paths.c

```diff
@@ -167,6 +169,10 @@ Path *AllPathsCtx_NextPath(AllPathsCtx *ctx) {
 				if(ctx->dst != NULL) {
 					Node dst = Path_Head(ctx->path);
 					if(ENTITY_GET_ID(ctx->dst) != ENTITY_GET_ID(&dst)) continue;
+					// if we're only collecting shortest paths,
+					// update the max level
+					if(ctx->shortest_paths && depth < ctx->maxLen)
+						ctx->maxLen = depth;
```

Comment:
What if the first path found between node `A` and `B` is of length 5, I'll assume that path is returned and processed, but the next path discovered between the two nodes is of length 3 ? isn't it a bit too late ? don't we need to know the min path length in advance ?

Reply:
You're right, I've added logic to find the min path length by BFS before starting to collect paths.